lib/apiservers/engine/backends/container.go

```diff
@@ -824,6 +828,21 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
 	return host, nil
 }
 
+func createNewAttachClientWithTimeouts(connectTimeout, responseTimeout, responseHeaderTimeout time.Duration) (*client.PortLayer, *httpclient.Transport) {
```

Comment:
This func takes three params that are never used. Looks like you should be using to define your timeouts when you create the transport on line 833.